kernel/thread.c

```diff
@@ -189,7 +189,7 @@ FUNC_NORETURN void _thread_entry(k_thread_entry_t entry,
 	_check_stack_sentinel();
 #endif
 #ifdef CONFIG_MULTITHREADING
-	k_thread_abort(_current);
+	k_thread_abort(k_current_get());
```

Comment:
We probably don't care (this only gets hit on thread exit) but this change introduces a double-hop into the kernel: we have to trap into k_current_get() come back to userspace, and then trap again into k_thread_abort().

A straightforward k_thread_self_abort() call could be implemetned at "k_thread_abort(_current)" and implemented with a single call.  Likely wouldn't even increase code size.

Reply:
I agree, but I'm not optimizing our interfaces right now.
I intend to do that around 1.11 development cycle, after all the infrastructure is in place.
There are many, many inefficiencies in our current APIs once the notion of system calls is introduced, but I feel it's premature at this time to dive into optimizing them.